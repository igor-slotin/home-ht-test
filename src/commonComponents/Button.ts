import styled from 'styled-components';

export const Button = styled.button`
  font-size: 16px;
  display: inline-block;
  padding: 0.5rem 1rem;
  border-radius: 2px;
  font-weight: normal;
  letter-spacing: 0.02em;
  cursor: pointer;
  color: #f1f1f1;
  border: none;
  background: #FD3B1A center;
  height: auto;
  min-height: 52px;
  min-width: 205px;
  width: 100%;
  transition: all .15s ease;

      &:not(:disabled) {
        outline: none;
        box-shadow: none;
      }
      &:hover, &:active {
       color: #ffffff;
       background: #FD3B1A;
       transform: translateY(-1px);
       box-shadow: 0 7px 14px rgba(50,50,93,.1), 0 3px 6px rgba(0,0,0,.08);
       
       transition: all .15s ease;
       .icon-right-chevron:before, .icon-left-chevron:before {
         color: #ffffff;
       }
       
      }
       &:focus, &:active:focus {
        outline: none;
        box-shadow: none;
        .icon-right-chevron:before {
         color: #ffffff;
       }
      }
     @media (max-width: 700px) {
        height: 2rem;
        font-size: 1rem;
        padding: 1rem;
        max-width: 80%;
        margin: 0 auto;
     }
`;