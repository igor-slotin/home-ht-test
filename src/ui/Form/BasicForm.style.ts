import styled from 'styled-components';
import { Progress } from 'react-sweet-progress';
import "react-sweet-progress/lib/style.css";

export const Container = styled.section`
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  max-width: 1300px;
  margin: 0 auto;
`

export const ContentWrapper = styled(Container)`
  justify-content: center;
  font-size: 2rem;
  height: calc(100% - 20px);
`

export const ProgressBar = styled(Progress)`
  @media (max-width: 700px) {
    position: fixed;
    bottom: 0;
  }
`