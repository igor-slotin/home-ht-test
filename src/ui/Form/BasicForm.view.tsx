import React, { Component } from 'react';
import { observer } from 'mobx-react'

import Store from '../../store/CollectedDataStore';

import { 
  Container,
  ContentWrapper,
  ProgressBar
} from './BasicForm.style'


@observer
export default class BasicFormView extends Component {
  componentDidMount() {
    Store.resetStep();
  }
  renderCurrentTab = () => {
    const CurrentTab = Store.currentTab.component;
    return CurrentTab && <CurrentTab />
  }
  render () {
    return (
      <Container>
        <ProgressBar percent={Store.progress} />
        <ContentWrapper>
          {this.renderCurrentTab()}
        </ContentWrapper>
      </Container>
    )
  }
}