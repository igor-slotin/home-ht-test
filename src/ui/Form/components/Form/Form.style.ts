import styled from 'styled-components';

export const Form = styled.form`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 200px;
`

export interface IProps {
  primary?: boolean;
  wide?: boolean;
  height?: boolean;
  smallBtn?: boolean;
  active?: boolean;
  editButton?:boolean;
}

export const Button = styled.button<IProps>`
  font-size: 16px;
  display: inline-block;
  padding: 0.5rem 1rem;
  border-radius: 2px;
  font-weight: normal;
  letter-spacing: 0.02em;
  cursor: pointer;
  color: '#FFFFFF'
  border: none;
  background: #FD3B1A center;
  height: auto;
  min-height: 52px;
  min-width: 205px;
  width: 100%;
  transition: all .15s ease;

      &:not(:disabled) {
        outline: none;
       box-shadow: none;
      }
      &:hover, &:active {
       color: #ffffff;
       background: #FD3B1A;
       transform: translateY(-1px);
       box-shadow: 0 7px 14px rgba(50,50,93,.1), 0 3px 6px rgba(0,0,0,.08);
       
       transition: all .15s ease;
       .icon-right-chevron:before, .icon-left-chevron:before {
         color: #ffffff;
       }
       
      }
       &:focus, &:active:focus {
        outline: none;
        box-shadow: none;
        .icon-right-chevron:before {
         color: #ffffff;
       }
      }
     @media screen and (min-width: 700px) {
        min-width: 205px;
        height: auto;
        padding: 1rem;
        max-width: 100%;
        margin: 0 auto;
     }
`;
