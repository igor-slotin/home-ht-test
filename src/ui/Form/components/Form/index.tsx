import FormView from './Form.view'
import Store from '../../../../store/CollectedDataStore'

export default (props) => (<FormView { ...props } store={Store}/>)