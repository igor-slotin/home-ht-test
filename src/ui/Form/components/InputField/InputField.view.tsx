import styled from 'styled-components';
const InputField = styled.input`
  padding: 10px;
  font-size: 16px;
  min-width: 205px;
  border: 1px solid #DDDDDD;
  margin: 20px 0;
  @media (max-width: 700px) {
    margin-top: 0;
    font-size: 1rem;
    margin-bottom: 1rem;
  }
`
export default InputField;