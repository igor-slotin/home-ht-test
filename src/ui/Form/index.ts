export { default } from './BasicForm.view';