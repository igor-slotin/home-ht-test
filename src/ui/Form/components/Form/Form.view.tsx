import React, { Component } from 'react'
import { observer } from 'mobx-react';

import { CollectedDataStore } from '../../../../store/CollectedDataStore'
import { Button } from '../../../../commonComponents/Button';

import { 
  Form,
 } from './Form.style'

 interface IProps {
   store: CollectedDataStore
 }

 @observer
export default class FormView extends Component<IProps> {
  render() {
    const { children, store } = this.props;
    return (
      <Form onSubmit={store.setNextTab} action="javascript:void(0);">
        {children}
        <Button type="submit">{store.isLastStep ? 'Submit params': 'Next step'}</Button>
      </Form>)
  }
}