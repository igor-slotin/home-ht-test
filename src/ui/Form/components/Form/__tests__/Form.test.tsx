import Form from '../Form.view';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import InputField from '../../InputField'
import { Button } from '../../../../../commonComponents/Button';
import { CollectedDataStore } from '../../../../../store/CollectedDataStore';

describe("Form test", () => {
  let snapshot;
  it('renders correctly', () => {
    const store = new CollectedDataStore();
    snapshot = renderer.create(
    <Form store={store}>
      <InputField/>
    </Form>)
  })

  it("Should match snapshot", () => {
    const tree = snapshot.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it ("Should have submit button", () => {
    const store = new CollectedDataStore();
    const formUi = shallow(<Form store={store}/>);
    expect(formUi.find(Button)).toHaveLength(1);
  })

  it ("Should have input element as a child button", () => {
    const store = new CollectedDataStore();
    const formUi = shallow(
    <Form store={store}>
      <InputField/>
    </Form>);
    expect(formUi.find(InputField)).toHaveLength(1);
  });

  it ('Should call set next tab action in store', () => {
    const store = new CollectedDataStore();
    const mockedFunction = jest.fn();
    store.setNextTab = mockedFunction;
    const formUi = mount(
      <Form store={store}>
        <InputField/>
      </Form>
    );
    formUi.find(Button).first().simulate('submit');
    expect(mockedFunction).toHaveBeenCalled();
  });

  it ('Button text should be "Next step"', () => {
    const store = new CollectedDataStore();
    const formUi = mount(
      <Form store={store}>
        <InputField/>
      </Form>
    );
    expect(formUi.find(Button).first().text()).toBe("Next step");
  });

  it ('Button text should be "Submit params"', () => {
    const store = new CollectedDataStore();
    store.isLastStep = true;
    const formUi = mount(
      <Form store={store}>
        <InputField/>
      </Form>
    );
    expect(formUi.find(Button).first().text()).toBe("Submit params");
  })
})