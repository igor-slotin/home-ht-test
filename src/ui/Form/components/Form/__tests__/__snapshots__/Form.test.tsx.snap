// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form test Should match snapshot 1`] = `
.c2 {
  font-size: 16px;
  display: inline-block;
  padding: 0.5rem 1rem;
  border-radius: 2px;
  font-weight: normal;
  -webkit-letter-spacing: 0.02em;
  -moz-letter-spacing: 0.02em;
  -ms-letter-spacing: 0.02em;
  letter-spacing: 0.02em;
  cursor: pointer;
  color: #f1f1f1;
  border: none;
  background: #FD3B1A center;
  height: auto;
  min-height: 52px;
  min-width: 205px;
  width: 100%;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c2:not(:disabled) {
  outline: none;
  box-shadow: none;
}

.c2:hover,
.c2:active {
  color: #ffffff;
  background: #FD3B1A;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
  box-shadow: 0 7px 14px rgba(50,50,93,.1),0 3px 6px rgba(0,0,0,.08);
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c2:hover .icon-right-chevron:before,
.c2:active .icon-right-chevron:before,
.c2:hover .icon-left-chevron:before,
.c2:active .icon-left-chevron:before {
  color: #ffffff;
}

.c2:focus,
.c2:active:focus {
  outline: none;
  box-shadow: none;
}

.c2:focus .icon-right-chevron:before,
.c2:active:focus .icon-right-chevron:before {
  color: #ffffff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: calc(100% - 20px);
}

.c1 {
  padding: 10px;
  font-size: 16px;
  min-width: 205px;
  border: 1px solid #DDDDDD;
  margin: 20px 0;
}

@media (max-width:700px) {
  .c2 {
    height: 2rem;
    font-size: 1rem;
    padding: 1rem;
    max-width: 80%;
    margin: 0 auto;
  }
}

@media (max-width:700px) {
  .c1 {
    margin-top: 0;
    font-size: 1rem;
    margin-bottom: 1rem;
  }
}

<form
  action="javascript:void(0);"
  className="c0"
  onSubmit={[Function]}
>
  <input
    className="c1"
  />
  <button
    className="c2"
    type="submit"
  >
    Next step
  </button>
</form>
`;
